ui/lib/config-ui/addon/routes/login-settings/rule/details.js

```diff
@@ -4,5 +4,26 @@
  */
 
 import Route from '@ember/routing/route';
+import { service } from '@ember/service';
 
-export default class LoginSettingsRuleDetailsRoute extends Route {}
+export default class LoginSettingsRuleDetailsRoute extends Route {
+  @service api;
+  @service store;
+
+  async model() {
+    const { name } = this.paramsFor('login-settings.rule');
+
+    const adapter = this.store.adapterFor('application');
+
+    const rule = await adapter.ajax(`/v1/sys/config/ui/login/default-auth/${encodeURI(name)}`, 'GET');
```

Comment:
ideally we wrap these kind of await's in a try/catch. Even if we swallow the error, noting that is helpful. I'm not super key'd into the flow here so maybe touch base with Claire or design for what should happen or be surfaced to the user if this fails.

Note too, sometimes it's helpful to pull out these async/awaits form the model hook an into their own function triggered in the model hook. It makes the flow a little cleaner.

[Here is an example you can work from. 
](https://github.com/hashicorp/vault/blob/main/ui/app/routes/vault/cluster/dashboard.js#L18)